daft/expressions/expressions.py

```diff
@@ -2442,6 +2442,58 @@ def truncate(self, interval: str, relative_to: Expression | None = None) -> Expr
         relative_to = Expression._to_expression(relative_to)
         return Expression._from_pyexpr(native.dt_truncate(self._expr, interval, relative_to._expr))
 
+    def to_string(self, format: str | None = None) -> Expression:
+        """Converts a datetime/date column to a string column. (alias for strftime)."""
+        return self.strftime(format=format)
```

Comment:
Do we need `to_string` vs casting as a string? I feel `strftime` makes sense for formatting with a **required argument** and `to_string` shouldn't take an arg so is just `cast(datetime as string)`. Long-term we might considering things like https://github.com/Eventual-Inc/Daft/issues/4014 which work on all types. This way we don't have many to/from methods for the builtin types. Then if there is some formatting arguments, these are separate format methods.

Reply:
so the `to_` method is purely for consistency with the inverse. (string -> date/datetime). We don't have a `.str.strptime`, but instead `to_` methods. 

- `.str.to_date(<fmt>)`
- `.str.to_datetime(<fmt>)`. 


This provides a similar experience for date to string conversions

Reply:
Yeah I saw this. What do you think of adding `strptime` for consistency with`strftime` then? Preferably we wouldn't have the to_date and to_datetime.

Reply:
i didnt want to deprecate/remove methods in this pr. Ill remove `to_string` and we can decide how to proceed with todate and tto_datetime later!